module/LearnZF2AjaxImageGallery/src/LearnZF2AjaxImageGallery/Controller/IndexController.php

```diff
@@ -107,10 +110,12 @@ protected function deleteimageAction()
             $data = $request->getPost()->toArray();
 
             if ($request->isXmlHttpRequest()) {
+                // @codeCoverageIgnoreStart
                 if (is_file("public".$data["img"])) {
                     unlink("public".$data["img"]);
                     $status = true;
                 }
+                // @codeCoverageIgnoreEnd
```

Comment:
So, by dedicated tests that extends `PHPUnit_Framework_TestCase`, you can do something like this:
```php
   $request   = $this->prophesize('Zend\Http\PhpEnvironment\Request');
   $request->isPost()->willReturn(true);
   
   $post = $this->prophesize('Zend\Stdlib\Parameters');
   $post->get('img')->willReturn('fileimage.jpg');

   $request->getPost()->willReturn($post);

   $request->isXmlHttpRequest()->willReturn(true);
   
   $r = new ReflectionProperty($this->controller, 'request');
   $r->setAccessible(true);
   $r->setValue($this->controller, $request->reveal());
  
    // do tweak for mkdir usage here
   $this->controler->deleteimageAction();
```